Migrate to `yarn workspaces`

I need to pilot this set up with this project before I decide what to do in generated projects. This came up in #74.

Nx prefers a single `package.json` file for the entire monorepo rather than multiple `package.json` files, one per project.

I find this odd because, particularly for publishing of shared libraries, the per-project `package.json` file is required, and would potentially need to be generated.

As is, there is duplication between the root `package.json` file and the project `package.json` file since dependencies must be listed in both in order to install properly for CI, and in order for published packages to include install of their own dependencies. 

(I just tried an experiment where I remove the dependencies from the project's `package.json`, and found that Nx generates a `package.json` that lists them as `peerDependencies` instead, which is _not_ what I want. I don't want client projects to care about what a package uses internally, on average.)

Obstacle here is that when I actually do switch to workspaces, E2E tests fail because for some reason `@chiubaka/nx-plugin:nx-plugin-e2e` suddenly can't be resolved.

A quick summary of the situation on this before I go grab lunch:

When I add the `workspaces` key to the root `package.json` and then re-run `yarn install`, E2E tests fail with this error:
```
> nx run nx-plugin-e2e:e2e


 >  NX   Cannot find module '/Users/dchiu/Developer/nx-plugin/packages/nx-plugin/src/executors/nxPluginE2e'

   Require stack:
   - /Users/dchiu/Developer/nx-plugin/node_modules/nx/src/config/workspaces.js
   - /Users/dchiu/Developer/nx-plugin/node_modules/nx/src/command-line/run.js
   - /Users/dchiu/Developer/nx-plugin/node_modules/nx/bin/run-executor.js
   Pass --verbose to see the stacktrace.
```

Removing `workspaces` key and re-running `yarn install` fixes the error, so I know that that's what is causing this.

Of course, this sequence has led me to search around for more info about using `yarn workspaces` with Nx. Naturally, I searched through the Nx issues for mention of yarn workspaces, and I have found a few kind of frightening issues that don't seem to have gotten any attention:
- https://github.com/nrwl/nx/issues/7115
- https://github.com/nrwl/nx/issues/1972

What this is showing is:
1. yarn workspaces maybe don't work as smoothly with Nx as advertised
2. Nx (or `nrwl` as a company?) has really failed to deliver on the idea that they would maintain this project well. There's so many issues and PRs that just get 0 attention whatsoever (I've even opened a couple of them and heard nothing thus far).

While I do think the actual issue at hand here is solvable, all of this has me worrying that further investment in Nx is a mistake. Some people in these comment threads are concluding the same thing...

The problem is mostly that Nx is very opinionated. It probably works great if you want to follow all of their recommendations. In my case, I'm definitely trying to rig Nx to do my own thing--I'm not interested in a single `package.json` file, and I'm definitely not interested in an org-level monorepo. I'm facing some pain around issues like this, and will undoubtedly face more pain in the future.

--- a/package.json
+++ b/package.json
@@ -19,6 +19,7 @@
     "lint:packages": "nx run-many --target=lint --all",
     "lint:root": "yarn eslint .",
     "lint:staged": "lint-staged",
+    "postinstall": "rm -rf ./node_modules/@chiubaka/nx-plugin",
     "prepare": "husky install",
     "start": "nx serve",
     "test": "nx test",
@@ -30,9 +31,7 @@
   "license": "UNLICENSED",
   "dependencies": {
     "@swc/helpers": "~0.3.3",
-    "tslib": "^2.3.0",
-    "winston": "^3.8.1",
-    "winston-transport": "^4.5.0"
+    "tslib": "^2.3.0"
   },
   "devDependencies": {
     "@chiubaka/eslint-config": "^0.3.7",
@@ -72,5 +71,10 @@
     "verdaccio": "^5.14.0",
     "verdaccio-auth-memory": "^10.2.0",
     "verdaccio-memory": "^10.3.0"
+  },
+  "workspaces": {
+    "packages": [
+      "packages/*"
+    ]
   }
 }

--- a/packages/nx-plugin/executors.json
+++ b/packages/nx-plugin/executors.json
@@ -4,7 +4,8 @@
     "nx-plugin-e2e": {
       "implementation": "./src/executors/nxPluginE2e",
       "schema": "./src/executors/nxPluginE2e/schema.json",
-      "description": "Sets up the E2E testing environment for an nx-plugin"
+      "description": "Sets up the E2E testing environment for an nx-plugin",
+      "cli": "nx"
     }
   }
 }

--- a/packages/nx-plugin/package.json
+++ b/packages/nx-plugin/package.json
@@ -7,13 +7,9 @@
     "deploy:ci": "yarn deploy"
   },
   "dependencies": {
+    "tslib": "^2.3.0",
     "winston": "^3.8.1",
     "winston-transport": "^4.5.0"
-  },
-  "devDependencies": {
-    "@types/eslint": "^8.4.5",
-    "@types/jest": "^28.1.6",
-    "jest": "^28.1.3"
   },
   "generators": "./generators.json",
   "executors": "./executors.json"

